webpack_loader/utils.py

```diff
@@ -21,12 +24,11 @@ def import_string(dotted_path):
         raise ImportError('%s doesn\'t look like a valid module path' % dotted_path)
 
 
-def get_loader(config_name):
-    if config_name not in _loaders:
-        config = load_config(config_name)
-        loader_class = import_string(config['LOADER_CLASS'])
-        _loaders[config_name] = loader_class(config_name, config)
-    return _loaders[config_name]
+@lru_cache(maxsize=None)
```

Comment:
This is changing the API w/o a clear benefit. While `_loaders` is private, someone might be using it.
I would refrain from changing this unless there's a very clear benefit.

Reply:
The 'very clear benefit' is to speed up config loading immensely, because template generation runs this function countless times when generating a single template, by looking at the various code paths.

`_loader` is a name that implies you shouldn't use that variable outside of the module, and it's everybody's responsibility to do just that, or face the consequences if they don't.

Reply:
The function already does the same thing lru_cache will do. _loaders is exactly the cache.

Reply:
Let me demonstrate you why reaching for a global variable is slower than using lru_cache that uses a cache table locally:
```python
import timeit
from functools import lru_cache
from random import randint

_some_global_variable = {}


def expensive_function(var1):
    return randint(0, 5)


def get1(var1):
    if var1 not in _some_global_variable:
        result = expensive_function(var1)
        _some_global_variable[var1] = result
        return result
    return _some_global_variable[var1]


@lru_cache(maxsize=None)
def get2(var1):
    return expensive_function(var1)


print(timeit.timeit(
    stmt='get1("x")', setup='from __main__ import get1', number=10000000))
print(timeit.timeit(
    stmt='get2("x")', setup='from __main__ import get2', number=10000000))
```

A couple run outputs with various python versions I have currently installed:

```
sh-5.2$ python3.8 test.py
0.7265332540009695
0.3478194800009078

sh-5.2$ python3.9 test.py
0.6371905300002254
0.3206007739972847

sh-5.2$ python3.10 test.py
0.48459655799888424
0.2498635900010413

sh-5.2$ python3.11 test.py
0.48944291500083636
0.38999983699977747

sh-5.2$ python3.12 test.py
0.425895505999506
0.3634855400014203
```